src/compiler/checker.ts

```diff
@@ -29937,17 +29937,21 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
             (node.kind === SyntaxKind.BinaryExpression && (node as BinaryExpression).operatorToken.kind === SyntaxKind.EqualsToken);
     }
 
+    function isSpreadIntoCallOrNew(node: ArrayLiteralExpression) {
+        const parent = walkUpParenthesizedExpressions(node.parent);
+        return isSpreadElement(parent) && isCallOrNewExpression(parent.parent);
+    }
+
     function checkArrayLiteral(node: ArrayLiteralExpression, checkMode: CheckMode | undefined, forceTuple: boolean | undefined): Type {
         const elements = node.elements;
         const elementCount = elements.length;
         const elementTypes: Type[] = [];
         const elementFlags: ElementFlags[] = [];
         pushCachedContextualType(node);
         const inDestructuringPattern = isAssignmentTarget(node);
-        const isSpreadIntoCallOrNew = isSpreadElement(node.parent) && isCallOrNewExpression(node.parent.parent);
-        const inConstContext = isSpreadIntoCallOrNew || isConstContext(node);

```

Comment:
While reading the source code today I noticed that `isSpreadIntoCallOrNew` here (as part of `inConstContext`).

I think it was semantically incorrect (and I added it [here](https://github.com/microsoft/TypeScript/pull/52845/files#diff-d9ab6589e714c71e657f601cf30ff51dfc607fc98419bf72e04f6b0fa92cc4b8R29669)). It didn't exactly create any issues because `isConstContext` here is mainly used to decide if the checked type should be readonly or not. 

Since I removed part of the condition we can now observe removal of the `readonly` modifier from some snapshot tests. I think it's fine... or at least, I wasn't yet able to prove that those tuples **should** be readonly. Even in actual `const` contexts etc a mutable array is still fine. Even if it turns out that the `readonly` modifier is needed for some case, I think that it should be added independently, based on `isSpreadIntoCallOrNew`.

As a bonus, I noticed that parenthesized expressions were not handled here so I improved this.

Reply:
The changed baselines that didn’t strictly need to change make this hard to approve. I’d rather not find out that the `readonly` was important in an untested edge case because of a regression.

Reply:
Should I split this into 2 PRs then? Note that [the change that allowed those](https://github.com/microsoft/TypeScript/pull/52845/files#diff-d9ab6589e714c71e657f601cf30ff51dfc607fc98419bf72e04f6b0fa92cc4b8R29669) and introduced those `readonly` modifiers in some cases were only merged into 5.1 so removing it really shouldn't affect many people.

Reply:
Ah, I missed that the same PR originally made them `readonly`. Probably fine then